code/modules/wiremod/components/sensors/view_sensor.dm

```diff
@@ -0,0 +1,56 @@
+/**
+ * # View Sensor
+ *
+ * Returns all movable objects in view.
+ */
+
+#define VIEW_SENSOR_RANGE 5
+#define VIEW_SENSOR_COOLDOWN 0.5 SECONDS
+
+/obj/item/circuit_component/view_sensor
+	display_name = "View Sensor"
+	desc = "Outputs a list with all movable objects in it's view. Requires a shell."
+	category = "Sensor"
+
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+	power_usage_per_input = 5 //Normal components have 1
```

Comment:
beef up power usage some more, ideally people shouldn't be spamming this component